libvast/src/system/index.cpp

```diff
@@ -428,143 +541,182 @@ caf::behavior index(caf::stateful_actor<index_state>* self, const path& dir,
+          if (iter->second.partitions.empty())
+            st.pending.erase(iter);
+        });
     },
     [=](atom::worker, caf::actor& worker) {
       self->state.idle_workers.emplace_back(std::move(worker));
     },
     [=](atom::done, uuid partition_id) {
-      self->state.decrement_indexer_count(partition_id);
+      // Nothing to do.
+      VAST_VERBOSE(self, "query for partition", partition_id, "is done");
     },
     [=](caf::stream<table_slice_ptr> in) {
       VAST_DEBUG(self, "got a new source");
       return self->state.stage->add_inbound_path(in);
     },
-    [=](atom::status, status_verbosity v) -> caf::config_value::dictionary {
-      return self->state.status(v);
+    [=](atom::status) -> caf::config_value::dictionary {
+      return self->state.status();
     },
-    [=](atom::subscribe, atom::flush, caf::actor& listener) {
+    [=](atom::subscribe, atom::flush, [[maybe_unused]] caf::actor& listener) {
       self->state.add_flush_listener(std::move(listener));
     });
-  return {[=](atom::worker, caf::actor& worker) {
-            auto& st = self->state;
-            st.idle_workers.emplace_back(std::move(worker));
-            self->become(caf::keep_behavior, st.has_worker);
-          },
-          [=](atom::done, uuid partition_id) {
-            self->state.decrement_indexer_count(partition_id);
-          },
-          [=](caf::stream<table_slice_ptr> in) {
-            VAST_DEBUG(self, "got a new source");
-            return self->state.stage->add_inbound_path(in);
-          },
-          [=](accountant_type accountant) {
-            namespace defs = defaults::system;
-            self->state.accountant = std::move(accountant);
-            self->send(self->state.accountant, atom::announce_v, "index");
-            self->delayed_send(self, defs::telemetry_rate, atom::telemetry_v);
-          },
-          [=](atom::status, status_verbosity v)
-            -> caf::config_value::dictionary { return self->state.status(v); },
-          [=](atom::subscribe, atom::flush, caf::actor& listener) {
-            self->state.add_flush_listener(std::move(listener));
-          }};
+  return {
+    // The default behaviour
+    [=](atom::worker, caf::actor& worker) {
+      auto& st = self->state;
+      st.idle_workers.emplace_back(std::move(worker));
+      self->become(caf::keep_behavior, st.has_worker);
+    },
+    [=](atom::done, uuid partition_id) {
+      VAST_VERBOSE(self, "query for partition", partition_id, "is done");
+    },
+    [=](caf::stream<table_slice_ptr> in) {
+      VAST_DEBUG(self, "got a new source");
+      return self->state.stage->add_inbound_path(in);
+    },
+    [=](accountant_type accountant) {
+      namespace defs = defaults::system;
+      self->state.accountant = std::move(accountant);
+    },
+    [=](atom::status) -> caf::config_value::dictionary {
+      return self->state.status();
+    },
+    [=](atom::subscribe, atom::flush, [[maybe_unused]] caf::actor& listener) {
```

Comment:
```suggestion
    [=](atom::subscribe, atom::flush, caf::actor& listener) {
```

Reply:
I'm also wondering why this is `actor&` instead of just `actor`.